### PR TITLE
[Refactor][Bug-Fix][Load Vec] Refactor code of basescanner and vjson/vparquet/vbroker scanner

### PR DESCRIPTION
**Pull request by HappenLee**
1. fix bug of vjson scanner not support `range_from_file_path`
2. fix bug of vjson/vbrocker scanner core dump by src/dest slot nullable is different
3. fix bug of vparquest filter_block reference of column in not 1
4. refactor code to simple all the code

# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/base_scanner.cpp [hunk 1]
```diff
@@ -71,6 +77,21 @@ Status BaseScanner::open() {
     _rows_read_counter = ADD_COUNTER(_profile, "RowsRead", TUnit::UNIT);
     _read_timer = ADD_TIMER(_profile, "TotalRawReadTime(*)");
     _materialize_timer = ADD_TIMER(_profile, "MaterializeTupleTime(*)");
+
+    const auto& range = _ranges[0];
```
Comment (cambyzju) [inline be/src/exec/base_scanner.cpp:81]: DCHECK(!_ranges.empty()); ?

File: be/src/exec/base_scanner.cpp [hunk 2]
```diff
@@ -272,59 +293,132 @@ Status BaseScanner::_fill_dest_tuple(Tuple* dest_tuple, MemPool* mem_pool) {
...
+    auto& filter_map = filter_column->get_data();
+
+    for (auto slot_desc : _dest_tuple_desc->slots()) {
+        if (!slot_desc->is_materialized()) {
+            continue;
+        }
+        int dest_index = ctx_idx++;
+
+        auto* ctx = _dest_vexpr_ctx[dest_index];
+        int result_column_id = 0;
```
Comment (cambyzju) [inline be/src/exec/base_scanner.cpp:329]:
```suggestion
        int result_column_id = -1;
```

File: be/src/vec/core/block.cpp [hunk 3]
```diff
@@ -402,6 +402,20 @@ std::string Block::dump_data(size_t begin, size_t row_limit) const {
     return out.str();
 }
 
+std::string Block::dump_one_line(size_t row, int column_end) const {
+    assert(column_end < columns());
+    fmt::memory_buffer line;
+    for (int i = 0; i < column_end; ++i) {
+        if (LIKELY(i != 0)) {
+            // TODO: need more effective function of to string. now the impl is slow
+            fmt::format_to(line, " {}", data[i].to_string(row));
```
Comment (carlvinhust2012) [inline be/src/vec/core/block.cpp:411]: why not check whether the data[i] is validity like this "  if (data[i].column) { s = data[i].to_string(row_num); } " ?

File: be/src/exec/base_scanner.cpp [hunk 4]
```diff
@@ -272,59 +293,132 @@ Status BaseScanner::_fill_dest_tuple(Tuple* dest_tuple, MemPool* mem_pool) {
...
+            for (int i = 0; i < rows; ++i) {
+                if (filter_map[i] && nullable_column->is_null_at(i)) {
+                    if (_strict_mode && (_src_slot_descs_order_by_dest[ctx_idx]) &&
+                        !_src_block.get_by_position(ctx_idx).column->is_null_at(i)) {
+                        RETURN_IF_ERROR(_state->append_error_msg_to_file(
+                                [&]() -> std::string {
+                                    return _src_block.dump_one_line(i, _num_of_columns_from_file);
+                                },
+                                [&]() -> std::string {
+                                    // Type of the slot is must be Varchar in _temp_block.
```
Comment (xiepengcheng01) [inline be/src/exec/base_scanner.cpp:346]: should modify this note?

File: be/src/exec/base_scanner.cpp (hunk 1 above)
Comment (HappenLee) [reply be/src/exec/base_scanner.cpp:81]: ok

File: be/src/vec/exec/vbroker_scanner.cpp [hunk 5]
```diff
@@ -44,18 +44,14 @@ VBrokerScanner::VBrokerScanner(RuntimeState* state, RuntimeProfile* profile,
     _text_converter.reset(new (std::nothrow) TextConverter('\\'));
 }
 
-VBrokerScanner::~VBrokerScanner() {}
+VBrokerScanner::~VBrokerScanner() = default;
```
[the remark this answers is not in the document]
Comment (HappenLee) [reply be/src/vec/exec/vbroker_scanner.cpp:47]: just do clang tidy clean

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**